File: maizecare_backend/src/models/SoilSensor.js
const mongoose = require('mongoose');

const soilSensorSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  plantId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Plant',
    required: false,
  },
  humidity: {
    type: Number,
    required: true,
    min: 0,
    max: 100,
  },
  temperature: {
    type: Number,
    required: false,
  },
  sensorId: {
    type: String,
    required: false,
  },
  location: {
    type: String,
    required: false,
  },
  recordedAt: {
    type: Date,
    default: Date.now,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('SoilSensor', soilSensorSchema);

File: maizecare_frontend/lib/data/models/soil_sensor.dart
class SoilSensor {
  final String id;
  final String userId;
  final String? plantId;
  final int humidity;
  final double? temperature;
  final String? sensorId;
  final String? location;
  final DateTime recordedAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  SoilSensor({
    required this.id,
    required this.userId,
    this.plantId,
    required this.humidity,
    this.temperature,
    this.sensorId,
    this.location,
    required this.recordedAt,
    required this.createdAt,
    required this.updatedAt,
  });

  factory SoilSensor.fromJson(Map<String, dynamic> json) {
    return SoilSensor(
      id: json['_id'] ?? '',
      userId: json['userId'] ?? '',
      plantId: json['plantId'],
      humidity: json['humidity']?.toInt() ?? 0,
      temperature: json['temperature']?.toDouble(),
      sensorId: json['sensorId'],
      location: json['location'],
      recordedAt: json['recordedAt'] != null
          ? DateTime.parse(json['recordedAt'])
          : DateTime.now(),
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'userId': userId,
      'plantId': plantId,
      'humidity': humidity,
      'temperature': temperature,
      'sensorId': sensorId,
      'location': location,
      'recordedAt': recordedAt.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
